### DISCUSSION TITLE
[discussion] Continued support for Python 2.7?

### DISCUSSION THREAD

**Discussion by miketheman** [UnitedIncome/serverless-python-requirements#600] [Ideas]
This plugin continues to support Python 2.7, which was [End Of Life as of Jan 1, 2020](https://www.python.org/dev/peps/pep-0373/).

There's plenty of references to Python 2.7 in the test suite and examples - we could cut down on some of the test suite runs if we remove those.

Appveyor CI is currently using Python2.7 as well - which likely needs updating, since it;s failing to install packages that are not supported in Python 2.7

I am considering working up a PR removing/updating references to remove Python 2.7 - however some might consider this a breaking change, necessitating a major version bump - however I don't believe there's anything breaking about removing _testing_ against that version. Looking for opinions here before moving forward.

**Comment by jimpudar** [comment]
I think removing 2.7 support is a fine idea, but I wouldn't remove the tests for 2.7 until the actual 2.7 functionality is removed and the major version is bumped.

I'm not personally relying on 2.7 support but I'd imagine many are, and removing tests for functionality people are still relying on sounds like a bad idea to me.

(Of course to play devil's advocate, using EOL software in the first place also seems like a bad idea to me.)

**Comment by rmax** [comment]
It's a tough call. If we assume AWS Lambda is the main target (do we have data for this?) the versions supported should follow what AWS Lambda currently supports: https://docs.aws.amazon.com/lambda/latest/dg/python-programming-model.html (2.7, 3.6+)

**Comment by bsamuel-ui** [comment]
Python 2.7 is [not scheduled for deprecation](https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html) on AWS.

I think, for clarity, it'd be best to release a new major version if we drop 2.7 support.

I'm assuming people still on 2.7 have old apps that aren't under active development and they just need to keep them running. In that scenario, it'd be clearest to set a constraint on the major version. After all, you're stuck on old versions of most libraries already...

Maybe we leave this open to hear if anyone has _active_ 2.7 apps, but it's hard to see why the answer isn't "update to 3 already."

**Comment by bsamuel-ui** [comment]
AWS is dropping support for 2.7 [in a few months](https://aws.amazon.com/blogs/compute/continued-support-for-python-2-7-on-aws-lambda/). I think we've reached a consensus that a major version release should drop support.

**Reply by bryantbiggs** [reply]
sounds great!

**Comment by bsamuel-ui** [comment]
[Draft PR to drop 2.7](https://github.com/UnitedIncome/serverless-python-requirements/pull/603).
